internal/gatewayapi/resource/resource.go

```diff
@@ -235,18 +225,266 @@ func (c *ControllerResources) DeepCopy() *ControllerResources {
 	return &out
 }
 
-// Equal implements the Comparable interface used by watchable.DeepEqual to skip unnecessary updates.
-func (c *ControllerResources) Equal(y *ControllerResources) bool {
-	// Deep copy to avoid modifying the original ordering.
-	c = c.DeepCopy()
-	c.sort()
-	y = y.DeepCopy()
-	y.sort()
-	return reflect.DeepEqual(c, y)
-}
-
-func (c *ControllerResources) sort() {
-	slices.SortFunc(*c, func(c1, c2 *Resources) int {
-		return cmp.Compare(c1.GatewayClass.Name, c2.GatewayClass.Name)
+func (c ControllerResources) Sort() {
+	// Top level sort based on gatewayClass contents
+	// Sort gatewayClass based on timestamp.
+	// Initially, sort by creation timestamp
+	// or sort alphabetically by “{namespace}/{name}” if multiple gatewayclasses share same timestamp.
+	sort.Slice(c, func(i, j int) bool {
+		if c[i].GatewayClass.CreationTimestamp.Equal(&(c[j].GatewayClass.CreationTimestamp)) {
+			return c[i].GatewayClass.Name < c[j].GatewayClass.Name
+		}
+		// Not identical CreationTimestamps
+		return c[i].GatewayClass.CreationTimestamp.Before(&(c[j].GatewayClass.CreationTimestamp))
 	})
+
+	// Then, run Sort for each item
+	for idx := range c {
+		// Sort gateways based on timestamp.
+		// Initially, gateways sort by creation timestamp
+		// or sort alphabetically by “{namespace}/{name}” if multiple gateways share same timestamp.
+		sort.Slice(c[idx].Gateways, func(i, j int) bool {
+			if c[idx].Gateways[i].CreationTimestamp.Equal(&(c[idx].Gateways[j].CreationTimestamp)) {
+				gatewayKeyI := fmt.Sprintf("%s/%s", c[idx].Gateways[i].Namespace, c[idx].Gateways[i].Name)
```

Comment:
would it be better to compare the namespace/name directly instead of doing a printf here?